### DISCUSSION TITLE
MessageCount in Thread

### DISCUSSION THREAD

**Discussion by ghost** [rabbitmq/rabbitmq-dotnet-client#1210] [Q&A] [answered]
Hello everyone,

I have tried to get the number of messages in the queue from the handler(Thread) but it always returns 0.

From the main thread it returns the amount.

I need to be able to know how many messages are left in the queue at any given time.

Thanks for the help

**Comment by michaelklishin** [comment, marked as answer]
You haven't provided a lot of detail on what exactly you did. If you did a passive queue declare, then it only will count messages in Ready state. So if you try one message at a time, or use automatic acknowledgement mode, those will typically be zero by the time your consumer handler is invoked.

**Comment by dapalex** [comment]
Hi,

I am having the same issue with RabbitMQ Client version 6.5.0 downloaded from nupkg.

My project is composed of a few containers which communicate each other through an event bus amqp-based (RabbitMQ).

The first container A is multithread working only as producer (it has a single shared connection among threads and one channel per thread). 
Each thread declares a durable queue using `IModel.QueueDeclare` method and send persistent messages.
The receiving container B (multiprocess) dequeues the messages from each queue (per process).

I downloaded the source code from github [here](https://github.com/rabbitmq/rabbitmq-dotnet-client/tree/v6.5.0) and ran quite a few tests.
While simply debugging with container A and the relating RabbitMQ container everything works fine: MessageCount returns the correct number of messages in the queue, running instead the whole project the issue occurs.
Playing with the containers I found out the issue occurs only when container B is subscribed to the queue as consumer: starting and stopping B, A consequently will show 0 messages when A is running and correct message count (thousands) when A stops.

Please let me know if further information is needed.

Thanks
